File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/nomenclator/NomenclatorAttributeSelectionFilterModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.nomenclator;

public class NomenclatorAttributeSelectionFilterModel {

	private Long id;
	private Long filterAttributeId;
	private String filterAttributeKey;
	private String defaultFilterValue;
	private String attributeKeyForAutocompleteFilterValue;
	
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public Long getFilterAttributeId() {
		return filterAttributeId;
	}
	public void setFilterAttributeId(Long filterAttributeId) {
		this.filterAttributeId = filterAttributeId;
	}
	public String getFilterAttributeKey() {
		return filterAttributeKey;
	}
	public void setFilterAttributeKey(String filterAttributeKey) {
		this.filterAttributeKey = filterAttributeKey;
	}
	public String getDefaultFilterValue() {
		return defaultFilterValue;
	}
	public void setDefaultFilterValue(String defaultFilterValue) {
		this.defaultFilterValue = defaultFilterValue;
	}
	public String getAttributeKeyForAutocompleteFilterValue() {
		return attributeKeyForAutocompleteFilterValue;
	}
	public void setAttributeKeyForAutocompleteFilterValue(String attributeKeyForAutocompleteFilterValue) {
		this.attributeKeyForAutocompleteFilterValue = attributeKeyForAutocompleteFilterValue;
	}
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/nomenclator/nomenclator-attribute-selection-filter.model.ts
import { JsonObject, JsonProperty } from "json2typescript";

@JsonObject
export class NomenclatorAttributeSelectionFilterModel {

	@JsonProperty("id", Number)
	public id: number = null;

	@JsonProperty("filterAttributeId", Number)
	public filterAttributeId: number = null;

	@JsonProperty("filterAttributeKey", String, true)
	public filterAttributeKey: string = null;

	@JsonProperty("defaultFilterValue", String, true)
	public defaultFilterValue: string = null;

	@JsonProperty("attributeKeyForAutocompleteFilterValue", String, true)
	public attributeKeyForAutocompleteFilterValue: string = null;
	
	public clone(): NomenclatorAttributeSelectionFilterModel {
		let clone: NomenclatorAttributeSelectionFilterModel = new NomenclatorAttributeSelectionFilterModel();
		Object.keys(this).forEach((prop: string) => {
			clone[prop] = this[prop];
		});
		return clone;
	}
}